platform/android/java/lib/src/org/godotengine/godot/GodotIO.java

```diff
@@ -345,6 +349,19 @@ public void dir_close(int id) {
 		//streams = new HashMap<Integer, AssetData>();
 		streams = new SparseArray<AssetData>();
 		dirs = new SparseArray<AssetDir>();
+		hasMulticastLock = false;
```

Comment:
You should be able to check ownership of the permission directly instead of using this workaround logic.
Check `org.godotengine.godot.utils.PermissionsUtil.java` for reference.